### PR TITLE
Update damping and time-step

### LINKED ISSUE #2574: Update damping and time-step options to atmos to support GFSv17 and GEFSv13
### What new functionality do you need?

Apply new damping and time-step options to all supported atmos resolutions and to both GFS and GEFS.
Apply the same damping to both GFS and GDAS cycles. Stronger damping was applied to GDAS in GFSv16.
This new damping option includes both explicit Rayleigh damping aaplied to u and v and implicit damping 
applied to w.  See this [document ](https://docs.google.com/document/d/1LOJ0Mip2ZXOHKQxIA_Pkv6GlTXUnOHQbFQl2Psuf2TQ/edit)for the options.  
This update makes the model more stable, allows longer time-step to be used for certain resolutions, and unify the damping options applied to GFS and GDAS cycles.

### What are the requirements for the new functionality?

Only requires workflow updates made to  parm/config/gefs/config.fcst parm/config/gefs/config.ufs parm/config/gfs/config.fcst parm/config/gfs/config.ufs.   No corresponding changes to ufs-weather-model are required. 

### Acceptance Criteria

pass workflow CI test

### Suggest a solution (optional)

_No response_

### PR DESCRIPTION
**Pull request by yangfanglin**
# Description
Updates the model to use explicit Rayleigh damping for u/v and implicit damping to w. This improves model stability and allows for longer timesteps. Also unifies the GDAS and GFS to use the same damping.  

Results from a test at the C1152 resolution (coupled model) can be found at https://www.emc.ncep.noaa.gov/gmb/wx24fy/C1152/newdamp/

Resolves #2574 

# Type of change
- New feature (adds functionality)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO

# How has this been tested?

# Checklist
- [ ] Any dependent changes have been merged and published
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

**Comment by yangfanglin** [comment]
@lisa-bengtsson  Lisa, would you like to add your following changes to branch https://github.com/yangfanglin/global-workflow/tree/dycore_update_hr4 and to this PR ?  

**xr_cnvcld = .false. for the C96 configuration only. It would need to be added both to config.ufs and to the parsing of the fv3 namelist. The default should be .true. for all other configurations.**

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]
Please review your code to make sure your indentation matches the surrounding code (using spaces) and remove the old code that has been commented out.

File: parm/config/gefs/config.ufs [hunk 1]
```diff
@@ -101,13 +106,18 @@ case "${fv3_res}" in
         export cdmbgwd="0.14,1.8,1.0,1.0"  # mountain blocking, ogwd, cgwd, cgwd src scaling
         export cdmbgwd_gsl="20.0,2.5,1.0,1.0"   # settings for GSL drag suite
         export knob_ugwp_tauamp=3.0e-3      # setting for UGWPv1 non-stationary GWD
+        export k_split=1
+	export n_split=4
+	export tau=8.0
+	export rf_cutoff=100.0
+        export fv_sg_adj=1800
```
Comment (WalterKolczynski-NOAA) [inline parm/config/gefs/config.ufs]: Missed one.
```suggestion
        export k_split=1
        export n_split=4
        export tau=8.0
        export rf_cutoff=100.0
        export fv_sg_adj=1800
```

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]
Looks fine other than one more indent fix.

**Review by WalterKolczynski-NOAA — APPROVED** [review]
Conditionally approved pending successful completion of CI tests

**Comment by lisa-bengtsson** [comment]
Hi Fanglin, yes I can add the xr_cnvcld namelist, let me see if I can do it from Orion as Hera is down today.

**Comment by CatherineThomas-NOAA** [comment]
I can confirm that the parameters in the gfs configs match the parameters specified in the [google doc](https://docs.google.com/document/d/1LOJ0Mip2ZXOHKQxIA_Pkv6GlTXUnOHQbFQl2Psuf2TQ/edit?usp=sharing).  I will wait for @lisa-bengtsson's update before formally approving.

**Comment by lisa-bengtsson** [comment]
I added a PR: https://github.com/yangfanglin/global-workflow/pull/1 
I believe Fanglin needs to review and merge?

File: parm/config/gefs/config.fcst [hunk 2]
```diff
@@ -107,19 +107,10 @@ if (( gwd_opt == 2 )); then
     export launch_level
 fi
 
-# Sponge layer settings
```
Comment (aerorahul) [inline parm/config/gefs/config.fcst:111]: Probably want to leave this comment, no?

File: parm/config/gfs/config.fcst [hunk 3]
```diff
@@ -119,19 +119,10 @@ if (( gwd_opt == 2 )); then
     export launch_level
 fi
 
-# Sponge layer settings
```
Comment (aerorahul) [inline parm/config/gfs/config.fcst:123]: Probably want to leave this comment, no?

**Review by aerorahul — APPROVED** [review]
Looks good. Just a couple of comments that were possibly accidentally removed.

File: parm/config/gefs/config.fcst (hunk 2 above)
Comment (yangfanglin) [reply parm/config/gefs/config.fcst:111]: @aerorahul you are right.  I will add this line back

**Comment by yangfanglin** [comment]
> I added a PR: [yangfanglin#1](https://github.com/yangfanglin/global-workflow/pull/1) I believe Fanglin needs to review and merge?

reviewed.  Need minor indentation adjustments

**Comment by lisa-bengtsson** [comment]
Sorry Fanglin, those indentations have a life on their own, maybe you have better luck on your end?

**Comment by yangfanglin** [comment]
> Sorry Fanglin, those indentations have a life on their own, maybe you have better luck on your end?

@WalterKolczynski-NOAA  Walter, I had the same trouble too but you were able to "suggest" changes for which I accepted.  Can you do the same to Lisa's modifications that I had already merged to this PR?  The indentations are still not aligned.

File: parm/config/gefs/config.ufs [hunk 4]
```diff
@@ -98,16 +104,22 @@ case "${fv3_res}" in
         export nthreads_fv3_gfs=1
         export nthreads_ufs=1
         export nthreads_ufs_gfs=1
-        export cdmbgwd="0.14,1.8,1.0,1.0"  # mountain blocking, ogwd, cgwd, cgwd src scaling
+	export xr_cnvcld=.false. # Do not pass conv. clouds to Xu-Randall cloud fraction
+	export cdmbgwd="0.14,1.8,1.0,1.0"  # mountain blocking, ogwd, cgwd, cgwd src scaling
```
Comment (aerorahul) [inline parm/config/gefs/config.ufs]:
```suggestion
	        export xr_cnvcld=".false."  # Do not pass conv. clouds to Xu-Randall cloud fraction
	        export cdmbgwd="0.14,1.8,1.0,1.0"  # mountain blocking, ogwd, cgwd, cgwd src scaling
```

File: parm/config/gfs/config.ufs [hunk 5]
```diff
@@ -96,9 +96,15 @@ case "${fv3_res}" in
         export nthreads_fv3_gfs=1
         export nthreads_ufs=1
         export nthreads_ufs_gfs=1
-        export cdmbgwd="0.071,2.1,1.0,1.0"  # mountain blocking, ogwd, cgwd, cgwd src scaling
+	export xr_cnvcld=.false. # Do not pass conv. clouds to Xu-Randall cloud fraction
+	export cdmbgwd="0.071,2.1,1.0,1.0"  # mountain blocking, ogwd, cgwd, cgwd src scaling
```
Comment (aerorahul) [inline parm/config/gfs/config.ufs]:
```suggestion
	        export xr_cnvcld=".false."  # Do not pass conv. clouds to Xu-Randall cloud fraction
	        export cdmbgwd="0.071,2.1,1.0,1.0"  # mountain blocking, ogwd, cgwd, cgwd src scaling
```

File: parm/config/gfs/config.ufs [hunk 6]
```diff
@@ -132,9 +138,15 @@ case "${fv3_res}" in
           export nthreads_fv3_gfs=1
           export nthreads_ufs=1
           export nthreads_ufs_gfs=1
-          export cdmbgwd="0.14,1.8,1.0,1.0"  # mountain blocking, ogwd, cgwd, cgwd src scaling
+	  export xr_cnvcld=.false. # Do not pass conv. clouds to Xu-Randall cloud fraction
+	  export cdmbgwd="0.14,1.8,1.0,1.0"  # mountain blocking, ogwd, cgwd, cgwd src scaling
```
Comment (aerorahul) [inline parm/config/gfs/config.ufs]:
```suggestion
	          export xr_cnvcld=.false. # Do not pass conv. clouds to Xu-Randall cloud fraction
	          export cdmbgwd="0.14,1.8,1.0,1.0"  # mountain blocking, ogwd, cgwd, cgwd src scaling
```

File: parm/config/gfs/config.ufs [hunk 7]
```diff
@@ -132,9 +138,15 @@ case "${fv3_res}" in
           export nthreads_fv3_gfs=1
           export nthreads_ufs=1
           export nthreads_ufs_gfs=1
-          export cdmbgwd="0.14,1.8,1.0,1.0"  # mountain blocking, ogwd, cgwd, cgwd src scaling
+	          export xr_cnvcld=.false. # Do not pass conv. clouds to Xu-Randall cloud fraction
+	          export cdmbgwd="0.14,1.8,1.0,1.0"  # mountain blocking, ogwd, cgwd, cgwd src scaling
```
Comment (aerorahul) [inline parm/config/gfs/config.ufs]:
```suggestion
	        export xr_cnvcld=.false. # Do not pass conv. clouds to Xu-Randall cloud fraction
	        export cdmbgwd="0.14,1.8,1.0,1.0"  # mountain blocking, ogwd, cgwd, cgwd src scaling
```

File: parm/config/gfs/config.ufs [hunk 8]
```diff
@@ -96,9 +96,15 @@ case "${fv3_res}" in
         export nthreads_fv3_gfs=1
         export nthreads_ufs=1
         export nthreads_ufs_gfs=1
-        export cdmbgwd="0.071,2.1,1.0,1.0"  # mountain blocking, ogwd, cgwd, cgwd src scaling
+	        export xr_cnvcld=".false."  # Do not pass conv. clouds to Xu-Randall cloud fraction
+	        export cdmbgwd="0.071,2.1,1.0,1.0"  # mountain blocking, ogwd, cgwd, cgwd src scaling
```
Comment (aerorahul) [inline parm/config/gfs/config.ufs]:
```suggestion
	      export xr_cnvcld=".false."  # Do not pass conv. clouds to Xu-Randall cloud fraction
	      export cdmbgwd="0.071,2.1,1.0,1.0"  # mountain blocking, ogwd, cgwd, cgwd src scaling
```

File: parm/config/gefs/config.ufs [hunk 9]
```diff
@@ -80,8 +80,14 @@ case "${fv3_res}" in
         export nthreads_fv3_gfs=1
         export nthreads_ufs=1
         export nthreads_ufs_gfs=1
-        export cdmbgwd="0.071,2.1,1.0,1.0"  # mountain blocking, ogwd, cgwd, cgwd src scaling
+	export xr_cnvcld=.false. # Do not pass conv. clouds to Xu-Randall cloud fraction
+	export cdmbgwd="0.071,2.1,1.0,1.0"  # mountain blocking, ogwd, cgwd, cgwd src scaling
```
Comment (aerorahul) [inline parm/config/gefs/config.ufs]:
```suggestion
	      export xr_cnvcld=.false. # Do not pass conv. clouds to Xu-Randall cloud fraction
	      export cdmbgwd="0.071,2.1,1.0,1.0"  # mountain blocking, ogwd, cgwd, cgwd src scaling
```

File: parm/config/gefs/config.ufs [hunk 10]
```diff
@@ -98,16 +104,22 @@ case "${fv3_res}" in
         export nthreads_fv3_gfs=1
         export nthreads_ufs=1
         export nthreads_ufs_gfs=1
-        export cdmbgwd="0.14,1.8,1.0,1.0"  # mountain blocking, ogwd, cgwd, cgwd src scaling
+	        export xr_cnvcld=".false."  # Do not pass conv. clouds to Xu-Randall cloud fraction
+	        export cdmbgwd="0.14,1.8,1.0,1.0"  # mountain blocking, ogwd, cgwd, cgwd src scaling
```
Comment (aerorahul) [inline parm/config/gefs/config.ufs]:
```suggestion
	      export xr_cnvcld=".false."  # Do not pass conv. clouds to Xu-Randall cloud fraction
	      export cdmbgwd="0.14,1.8,1.0,1.0"  # mountain blocking, ogwd, cgwd, cgwd src scaling
```

**Review by aerorahul — APPROVED** [review]
looks good.

**Comment by lisa-bengtsson** [comment]
Thanks Rahul!

**Comment by yangfanglin** [comment]
@CatherineThomas-NOAA  now ready for you to review

**Comment by XiaqiongZhou-NOAA** [comment]
It looks good to me. Just wondering why some parameters are in config.fcst and some in config.ufs.

**Comment by yangfanglin** [comment]
> It looks good to me. Just wondering why some parameters are in config.fcst and some in config.ufs.

@XiaqiongZhou-NOAA   parameters that need to be set differently for different model resolutions are usually included in config.ufs

**Review by CatherineThomas-NOAA — APPROVED** [review]
I ran our C384 3DVar case with the new config files and confirmed that the parameters are set correctly in the forecast step.  I also checked the GSI omf stats for that cycle and fits are reasonable.

Looks good to me.  Approve.

**Review by KevinViner-NOAA — APPROVED** [review]

**Comment by yangfanglin** [comment]
@WalterKolczynski-NOAA @aerorahul  ready to run CI tests ?

**Comment by aerorahul** [comment]
> @WalterKolczynski-NOAA @aerorahul ready to run CI tests ?

Yes.  There is a queue. Should be soon.

**Review by XiaqiongZhou-NOAA — APPROVED** [review]

**Comment by WalterKolczynski-NOAA** [comment]
Looks like something is wrong with your namelist.

**Comment by yangfanglin** [comment]
I checked the ufs-weather-model found at /scratch1/NCEPDEV/global/CI/2575/gfs/sorc/ufs_model.fd, the namelist parameter xr_cnvcld cannot be found in any of the source code under ./FV3/ccpp  @lisa-bengtsson has your code PR been merged to ufs-weather-model ?

**Comment by lisa-bengtsson** [comment]
You would need this update from March: In order to use this namelist flag the version of the ufs-weather-model needs to include this update: https://github.com/ufs-community/ufs-weather-model/pull/2202

**Comment by WalterKolczynski-NOAA** [comment]
> You would need this update from March: In order to use this namelist flag the version of the ufs-weather-model needs to include this update: [ufs-community/ufs-weather-model#2202](https://github.com/ufs-community/ufs-weather-model/pull/2202)

Figures; we are using the UFS commit directly before that. We are unable to update UFS to current because of issues writing through symlinks in CICE, but I think you can update the model hash in this PR just to the version that includes your changes (`c54e986`). 

To update a submodule, recursively clone workflow, then go into the `sorc/ufs_model.fd` directory and checkout the desired version. Then return up to the workflow and add the new version to a commit using `git add sorc/ufs_model.fd`. Then commit and push.

**Comment by yangfanglin** [comment]
@WalterKolczynski-NOAA will moving to the later  hash with Lisa's commit causing trouble for any other applications ?

**Comment by lisa-bengtsson** [comment]
> @WalterKolczynski-NOAA will moving to the later hash with Lisa's commit causing trouble for any other applications ?

Do you mean besides the xr_cnvcld update? 

The xr_cnvcld is set to true default. It did not change baselines in the ufs-weather-model RT's. In the workflow I use config.ufs to set it to false for the coarse resolution, so it will not impact HR4, but it will impact SFS C96 tests. (Which is the purpose of this update).

**Comment by yangfanglin** [comment]
@lisa-bengtsson yes,   besides the xr_cnvcld update.

**Comment by WalterKolczynski-NOAA** [comment]
If you go forward just the one commit, I think things will be fine (as long as there aren't other namelist changes you haven't included), but that's also why we test.

**Comment by lisa-bengtsson** [comment]
There were no baseline changes in the ufs RT's in this commit which is only one ahead of current, so that may be a good thing. https://github.com/ufs-community/ufs-weather-model/pull/2202

**Comment by yangfanglin** [comment]
@lisa-bengtsson   Which ufs-weather-model hash contains your xr_cnvcld  update ?   "git log" at ufs_model.fd does not show any relevant info.

commit 7fdb58cad0dad2f62ce7813c6719554d1c5a17af
Author: RatkoVasic-NOAA <37597874+RatkoVasic-NOAA@users.noreply.github.com>
Date:   Fri Mar 22 09:09:38 2024 -0400

    Fix modulefiles for Hera/Rocky8 OS. (#2194)

    - update module paths in Hera intel/gnu lua files for Rocky8 OS spack-stack libraries.

commit 3c8338c1436912e4547c3a2db7ced9dd7b70e782
Author: Dusan Jovic <48258889+DusanJovic-NOAA@users.noreply.github.com>
Date:   Tue Mar 19 20:55:14 2024 -0400

    Fix type mismatch compiler error when gfortran 10 is used without '-fallow-argument-mismatch' flag (#1147)

    * UFSWM - Fix type mismatch compiler error when gfortran 10 is used without '-fallow-argument-mismatch' flag.
      * FV3 - Fix type mismatch compiler error when gfortran 10 is used without '-fallow-argument-mismatch' flag.
        * ccpp-physics - Resolve various subroutine argument mismatches.
        * ccpp-framework - Add support to use mpi_f08 MPI module .
      * stochastic_physics - Fix type mismatch compiler error when gfortran 10 is used without '-fallow-argument-mismatch' flag.

commit 86b777354252f655cc6c9a72dba3fef48fcc8b80
Author: Dom Heinzeller <dom.heinzeller@icloud.com>
Date:   Mon Mar 18 12:44:52 2024 -0600

    Updates of ccpp-framework and ccpp-physics (merge ccpp-framework feature/capgen into main/20240308) Combined PR #2190 (#2181)

    * FV3 - Update submodule pointers for ccpp-framework and ccpp-physics. Change units flashes 5 min-1 to flashes min-1 and update long name to make clear this is per 5 minutes.
        * ccpp-physics - In physics/Interstitials/UFS_SCM_NEPTUNE/maximum_hourly_diagnostics.meta: change units flashes 5 min-1 to flashes min-1 and update long name to make clear this is per 5 minutes.
        * ccpp-framework - Update main from feature/capgen as of 2024-03-08 (includes optional argument updates in feature/capgen). Only commit on top of the merge is https://github.com/NCAR/ccpp-framework/commit/6cdd38a032d89d0f79c66f37f0c7172e59f135cd which is required to parse the metadata in the Unified Forecast System / ccpp-physics (underscores in units, e.g. degree_north).

commit e4961eb278ac0eb4d8f672851092e2cd4d072e0d
Author: Grant Firl <grant.firl@noaa.gov>
Date:   Thu Mar 14 16:38:41 2024 -0400

    Incorporate csawMG update and pass both intel and GNU RT tests (#2180)

    * UFSWM - Update the CS convection scheme
      * FV3 - Update the CS convection scheme
        * ccpp-physics - Update the CS convection scheme

commit 126db7a5145319cb6a8b3d272215efcd5b88562f
Author: Denise Worthen <denise.worthen@noaa.gov>
Date:   Tue Mar 12 15:20:34 2024 -0400

    Remove compiler warnings for CMEPS and CICE; Add traditional threading templates for global apps and append _esmf for templates using ESMF-managed threading (was #2179) (#2172)

    * CICE and CMEPS - Remove compiler warnings.
    * UFSWM - Allow flexibility to use traditional or ESMF-managed threading for global apps.

commit 5b62e1aa2e67ea58680d58ab16264d69b4085ea8
Author: Dom Heinzeller <dom.heinzeller@icloud.com>
Date:   Mon Mar 11 10:43:08 2024 -0600

    Add conflict for MOVING_NEST option with GNU Fortran, use mcmodel=small on macOS arm64 (also contains #2017) (#2164)

    * UFSWM - "Add conflict for MOVING_NEST option with GNU Fortran, use mcmodel=small on macOS arm64"
       * FV3 - "Bug fix for unallocated array sfcprop%wetness in      GFS_diagnostics.F90"
         * ccpp-physics -
         * ccpp-framework - "Bug fix required to use chunked arrays and retain backward compatibility with blocked data
            structures in ccpp_prebuild.py"
       * Creation of UFS-WM jenkinsfile for pipeline integration
       * Jet Rocky8 updates
         * spack-stack module path
         * eflow location and version

commit 458486087147216a9154f2169f8a338ff747943c
Author: jiandewang <jiande.wang@noaa.gov>
Date:   Mon Mar 4 17:25:05 2024 -0500

    update MOM6 to its main repo. 20240228 updating (#2166)

    * MOM6 -
    update MOM6 to its main repo. 20240228 updating (which is the original GFDL 20231113 candidate PR)

commit 8518c2c59fd92ec8a5982997d66612992b577dda
Author: Grant Firl <grant.firl@noaa.gov>
Date:   Fri Mar 1 11:43:59 2024 -0500

    tisfc bugfix (#2137)
.........................

**Comment by JessicaMeixner-NOAA** [comment]
@yangfanglin I think it's this commit hash: 
https://github.com/ufs-community/ufs-weather-model/commit/c54e98637ead81b1fc1e336bd0443c8bfb6faf01

**Comment by lisa-bengtsson** [comment]
Can you try git checkout c54e986 ?

The commit message should be from March 27th and look like this:

add xr_cnvcld flag to FV3 (https://github.com/ufs-community/ufs-weather-model/pull/2185) (https://github.com/ufs-community/ufs-weather-model/pull/2202)

* UFSWM - atparse.bash: correctly handle input that doesn't end with an end-of-line character. Fix some bugs in Rocoto support and clean up rt.sh.
  * FV3 - namelist flag xr_cnvcld to control if suspended grid-mean convective cloud condensate should be included in cloud fraction and optical depth calculation in radiation in the GFS suite
    * ccpp - physics-level changes to implement new namelist variable
  * MOM6 - update MOM6 code to eliminate all compiler warnings

**Comment by lisa-bengtsson** [comment]
If you need the full hash, it is: c54e98637ead81b1fc1e336bd0443c8bfb6faf01

**Comment by yangfanglin** [comment]
doing coding between meetings .......

Now I think I have everything figured out.   Please restart the CI test

**Comment by yangfanglin** [comment]
It is strange why the CI tests failed on Hera and Orion but all worked on WCOSS2.  The log files from failed cases showed the same namelist errors before changing the model hash.  Was the model recompiled on Orion and Hera for the CI tests ?  Do the old CI test directories need to be removed before  restarting the CI tests on HERA and ORION ?

**Comment by WalterKolczynski-NOAA** [comment]
> It is strange why the CI tests failed on Hera and Orion but all worked on WCOSS2. The log files from failed cases showed the same namelist errors before changing the model hash. Was the model recompiled on Orion and Hera for the CI tests ? Do the old CI test directories need to be removed before restarting the CI tests on HERA and ORION ?

It should have been, but I will ask @TerrenceMcGuinness-NOAA to confirm.

**Comment by WalterKolczynski-NOAA** [comment]
We're re-running again after making a CI tweak.

**Comment by WalterKolczynski-NOAA** [comment]
I'm trying to sort out what the issue is on Orion. Just ignore them for now.

**Comment by JessicaMeixner-NOAA** [comment]
I just looked at this orion failure, it looks a bit odd.  Do we think this is a machine issue or something else?

**Comment by yangfanglin** [comment]
One of the job failed at the post processing step for FH71 but the forecast started from FH78.  It is strange.  It appears the job was rerun as a warm start but this group of post job still tried to processing earlier forecast hours.   Can someone remove everything and restart the CI test on Orion ?

**Review by JessicaMeixner-NOAA — APPROVED** [review]

**Pull request merged**